events/events.go

```diff
@@ -0,0 +1,64 @@
+package events
+
+import (
+	"context"
+	//"fmt"
+
+	"github.com/flanksource/commons/logger"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/flanksource/incident-commander/responder"
+)
+
+func ProcessQueue() error {
+
+	ctx := context.Background()
+
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return err
+	}
+
+	_, err = conn.Exec(ctx, "LISTEN event_queue_updates")
+	if err != nil {
+		return err
+	}
+
+	logger.Infof("Started event processor")
+
+	// Reconcile pending events
+	reconcileEvents()
+
+	for {
+		_, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			return err
+		}
+
+		reconcileEvents()
+
+	}
+}
+
+func reconcileEvents() error {
+	// TODO: Add attempts where clause
+	var events []api.Event
+	tx := db.Gorm.Raw("SELECT id, properties FROM event_queue FOR UPDATE SKIP LOCKED").Scan(&events)
```

Comment:
I think this needs to be done inside of a db transaction to work correctly, Without the transaction - the lock will be released in the next line.

Reply:
It also needs to `limit 1` otherwise if there is an error processing a single item, then the entire transaction for the batch is rolled bacl. 

I would maybe put this into a loop, that selects, if nothing found then sleep for configurable time period, and then  use a channel to wake from the sleep 

```go
select {
	case m := <-pgNotify:
		handle(m)
	case <-time.After(10 * time.Second):
		fmt.Println("timed out")
	}
```